### PR TITLE
Fix Issue 1892. Client-sdk: setting invalid security level/hash function family leave the chain object in a corrupted state

### LINKED ISSUE #1892: Client-sdk: setting invalid security level/hash function family leave the chain object in a corrupted state

### PR DESCRIPTION
**Pull request by adecaro**
## Description

This PR introduces the following modification: Before modifying the security level and/or the hash algorithm on the MemberServices object, the underlying implementation checks that indeed the passes values respect the constraints, namely for the security level only 256 and 384 are allowed and for the hash family on 'SHA2' and 'SHA3' is allowed. This avoid that the MemberServices remains in an invalid state.
## Motivation and Context

Fixes #1892
## How Has This Been Tested?

A new test has been introduced in chain-tests.js
## Checklist:

- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff).
- [x] I have either added documentation to cover my changes or this change requires no new documentation.
- [x] I have either added unit tests to cover my changes or this change requires no new tests.
- [] I have run [golint](https://github.com/golang/lint) and have fixed valid warnings in code I have added or modified. This tool generates false positives so you may choose to ignore some warnings. The goal is clean, consistent, and readable code.

The continuous integration build process will run [make checks](https://github.com/hyperledger/fabric/blob/master/Makefile#L22) to confirm that tests pass and that code quality meets minimum standards. You may optionally run this locally as PRs will not be accepted until they pass.

Signed-off-by: Angelo De Caro adc@zurich.ibm.com

### CODE REVIEW

**Comment by srderson** [comment]
@smithbk or @angrbrd - please review

**Comment by smithbk** [comment]
LGTM

**Comment by srderson** [comment]
Thanks for the more detailed description in the PR. Issue #1892 does not contain enough info. LGTM

File: common/ledger/blkstorage/test_exports.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package blkstorage
+
```
Comment (manish-sethi) [inline common/ledger/blkstorage/test_exports.go]: As this file is mainly for providing other package with a help, it is cleaner to have a separate package for this. From what we have used at other places [1](https://github.com/hyperledger/fabric/tree/master/core/ledger/ledgermgmt/ledgermgmttest), and [2](https://github.com/hyperledger/fabric/tree/master/core/ledger/confighistory/confighistorytest), following the pattern from standard library [httptest](https://github.com/golang/go/tree/master/src/net/http/httptest), the name of the package could be - `blkstorage/blkstoragetest`.

**Pull request merged**